src/ripple/ledger/impl/View.cpp

```diff
@@ -1264,22 +1065,34 @@ offerDelete (ApplyView& view,
     auto owner = sle->getAccountID  (sfAccount);
 
     // Detect legacy directories.
-    bool bOwnerNode = sle->isFieldPresent (sfOwnerNode);
     uint256 uDirectory = sle->getFieldH256 (sfBookDirectory);
 
-    TER terResult  = dirDelete (view, false, sle->getFieldU64 (sfOwnerNode),
-        keylet::ownerDir (owner), offerIndex, false, !bOwnerNode, j);
-    TER terResult2 = dirDelete (view, false, sle->getFieldU64 (sfBookNode),
-        keylet::page (uDirectory), offerIndex, true, false, j);
+    bool const deletedOwner = view.dirRemove(
+        keylet::ownerDir(owner),
+        sle->getFieldU64(sfOwnerNode),
+        offerIndex,
+        false);
```

Comment:
This seems fine as is, but I think we could also do what we're doing elsewhere and return `tefBAD_LEDGER` right away if a `dirRemove` fails.

Reply:
Right, I've just made the change so that we just immediately remove `tefBAD_LEDGER` if a `dirRemove` fails. It would be a bit awkward to not be consistent here.